compiler/src/dmd/cparse.d

```diff
@@ -2168,6 +2168,7 @@ final class CParser(AST) : Parser!AST
      * C11 Initialization
      * initializer:
      *    assignment-expression
+     *    { }                       // C23 6.7.10 addition
```

Comment:
I understood C23 syntax to be `int var = ();` - this needs to be checked again.

Reply:
It's `{ }`. See C23 6.7.10